tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -317,3 +296,94 @@ def test_munge_flutter_worked(self, mock_integration):
+            stack_root="",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by automation, not as well defined stack root
+        self.code_mapping2 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/src/",
+            source_root="",
+            automatically_generated=True,
+        )
+        # Created by the user, well defined stack root
+        self.code_mapping3 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by the user, not as well defined stack root
+        self.code_mapping4 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by automation, well defined stack root
+        self.code_mapping5 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/src/sentry/",
+            source_root="",
+            automatically_generated=True,
+        )
+        self.code_mappings = [
+            self.code_mapping1,
+            self.code_mapping2,
+            self.code_mapping3,
+            self.code_mapping4,
+            self.code_mapping5,
+        ]
+
+        self.filepath = "usr/src/getsentry/src/sentry/src/sentry/utils/safe.py"
+
+    def test_test_multiple_code_mapping_matches_order(self):
+        project_stacktrace_link_endpoint = ProjectStacktraceLinkEndpoint()
+
+        configs = self.code_mappings
+        # Expected configs: stack_root, automatically_generated
+        expected_config_order = [
+            self.code_mapping3,  # "usr/src/getsentry/", False
+            self.code_mapping4,  # "usr/src/", False
+            self.code_mapping1,  # "", False
+            self.code_mapping5,  # "usr/src/getsentry/src/sentry/", True
+            self.code_mapping2,  # "usr/src/getsentry/src/", True
+        ]
+        sorted_configs = project_stacktrace_link_endpoint.sort_code_mapping_configs(configs)
+        assert sorted_configs == expected_config_order
+
+    def test_multiple_code_mapping_matches(self):
+        with mock.patch.object(
+            ExampleIntegration,
+            "get_stacktrace_link",
+            return_value="https://github.com/usr/src/getsentry/src/sentry/src/sentry/utils/safe.py",
+        ):
+            response = self.get_success_response(
+                self.organization.slug, self.project.slug, qs_params={"file": self.filepath}
```

Comment:
There's multiple matches because `usr/src/getsentry/src/sentry/src/sentry/utils/safe.py` matches the following stack roots: `usr/src/getsentry/` and `usr/src/`.